### PR TITLE
fix for #220

### LINKED ISSUE #220: Archive component
**Is your feature request related to a problem? Please describe.**

According to the `Zest_Framework/src/Zip/`, it looks like the Archive only supports the `Zip`.

But it has other archive approaches. For example, the `bzip`, `gzip` and so on.

**Describe the solution you'd like**

Consider creating missed archive components.

Or try considering creating a `Archive` component to support all available archives in this framework.

### PR DESCRIPTION
**Pull request by lablnet**
https://github.com/zestframework/Zest_Framework/issues/220

### CODE REVIEW

**Review by peter279k — CHANGES_REQUESTED** [review]
These classes seems to be done and my recommendations are as follows:

- Each archive classes should have their unit tests and it can make their behaviors verified and reliable.
- The StyleCI has been failed to check on the latest commit and it should be fixed.

**Comment by lablnet** [comment]
@Maikuolan tests pass in `gh actions`

**Review by Maikuolan — APPROVED** [review]
LGTM. :-)

**Comment by peter279k** [comment]
coding style [error ](https://github.styleci.io/analyses/yv0QnN?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)should be concerned.

File: src/Archive/Archive.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    /**
+     * Set the adapter.
+     *
+     * @param (string) $adapter
+     *
+     * @since 3.0.0
+     *
+     * @return object
+     */
+    public function setAdapter($adapter)
```
Comment (peter279k) [inline src/Archive/Archive.php]: To be consistency, it should add the native type hint:

```PHP
public function setAdapter($adapter): self
```

**Comment by lablnet** [comment]
> coding style [error ](https://github.styleci.io/analyses/yv0QnN?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)should be concerned.

there is no such error, its giving that in other file which is out of scope in sense of current PR

**Comment by peter279k** [comment]
> > coding style [error ](https://github.styleci.io/analyses/yv0QnN?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)should be concerned.
> 
> there is no such error, its giving that in other file which is out of scope in sense of current PR

Right. Without coding style error, I think it should be good to merge this PR.

**Pull request merged**